test/integration/targets/aws_ses_identity_dkim/tasks/main.yml

```diff
@@ -0,0 +1,172 @@
+
+    ## DKIM enable idempotence test (rerun)
+
+    - name: Enable DKIM verification on domain identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert no change (itempotence)
+      assert:
+        that:
+          - domain_result.changed == False
+          - domain_result.dkim_attributes.dkim_enabled == True
+
+    - name: Enable DKIM verification on email identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert no change (idempotence)
+      assert:
+        that:
+          - email_result.changed == False
+          - email_result.dkim_attributes.dkim_enabled == True
+
+    ## DKIM disable test
+
+    - name: Disable DKIM verification on domain identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - domain_result.changed == True
+          - domain_result.dkim_attributes.dkim_enabled == False
+
+    - name: Disable DKIM verification on email identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - email_result.changed == True
+          - email_result.dkim_attributes.dkim_enabled == False
+
+    ## DKIM disable idempotence test (rerun)
+
+    - name: Disable DKIM verification on domain identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - domain_result.changed == False
+          - domain_result.dkim_attributes.dkim_enabled == False
+
+    - name: Disable DKIM verification on email identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - email_result.changed == False
+          - email_result.dkim_attributes.dkim_enabled == False
+
+    ## DKIM check mode test (nothing should change, dry-run)
+
+    - name: Enable DKIM verification on email identity (check-mode)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: disabled
+        <<: *aws_connection_info
+      check_mode: True
+      register: email_result
+
+    - name: assert results being unchanged (check-mode)
```

Comment:
This test passes even if it's not run in check mode, so it's not ideal for testing check_mode

It would be better to test the check mode before the eable / after disable as that will fail if check mode actually makes the changes.

I've created this approach here if you want to just grab it: https://github.com/orthanc/ansible/commit/1cfa2166c868d718fd9a43693623f5499a538713